src/main/java/seedu/address/model/person/Person.java

```diff
@@ -18,23 +16,23 @@ public class Person {
 
     // Identity fields
     private final Name name;
-    private final Phone phone;
+    private final ClassGroup classGroup;
     private final Email email;
-
-    // Data fields
-    private final Address address;
-    private final Set<Tag> tags = new HashSet<>();
+    private final Phone phone;
+    private final Optional<Telegram> telegram;
+    private final Optional<Github> github;
 
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, tags);
+    public Person(Name name, ClassGroup classGroup, Email email, Phone phone, Telegram telegram, Github github) {
+        requireAllNonNull(name, classGroup, email, phone);
```

Comment:
Since we allow optional telegram and github, is it better to change the constructor to take in Optional<> of Telegram and Github ?

Otherwise the Optional.ofNullable(telegram) feels pointless since that part of code is unreachable due to `requireAllNonNull`. But I think an Optional.empty would pass this non null test because its only checking for a `null`.

Reply:
yep, and ive also changed the parseTele and parseGithub to return Optional wrapped instead to simplify this